src/test_ivio/bcf_mt_reader.cpp

```diff
@@ -0,0 +1,207 @@
+        ivio::bcf::record{.chromId = 0, .pos = 1230236, .rlen = 1, .qual = 47, .n_info = 3, .n_allele = 1, .n_sample = 3, .n_fmt = 4, .id = "",          .ref = "T",   .alt = {},                           .filter = {17, 0}, .info = {17,  1, 17,  3, 17,  2, 17, 13, 17,  4, 23, 84},                                                               .format = {17,  9, 33,  2,  3,  2,  3,  2,  2, 17, 10, 17, 54, 48, 61, 17,  2, 17,  7,  4,  2, 17, 11, 33, 56, 60, 51, 51, 128, 129}},
+        ivio::bcf::record{.chromId = 0, .pos = 1234566, .rlen = 3, .qual = 50, .n_info = 3, .n_allele = 3, .n_sample = 3, .n_fmt = 3, .id = "microsat1", .ref = "GTC", .alt = {23, 71, 71, 71, 84, 67, 84}, .filter = {17, 0}, .info = {17,  1, 17,  3, 17,  2, 17,  9, 17,  4, 23, 71},                                                               .format = {17,  9, 33,  2,  4,  2,  6,  4,  4, 17, 10, 17, 35, 17, 40, 17,  2, 17,  4,  2,  3}},
+    };
+
+    auto expected_header = ivio::bcf::header {
+        .table = {
+            {R"(fileformat)", R"(VCFv4.3)"},
+            {R"(FILTER)", R"(<ID=PASS,Description="All filters passed",IDX=0>)"},
+            {R"(fileDate)", R"(20090805)"},
+            {R"(source)", R"(myImputationProgramV3.1)"},
+            {R"(reference)", R"(file:///seq/references/1000GenomesPilot-NCBI36.fasta)"},
+            {R"(contig)", R"(<ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e0c7f379d618ff66beb2da,species="Homo sapiens",taxonomy=x,IDX=0>)"},
+            {R"(phasing)", R"(partial)"},
+            {R"(INFO)", R"(<ID=NS,Number=1,Type=Integer,Description="Number of Samples With Data",IDX=1>)"},
+            {R"(INFO)", R"(<ID=DP,Number=1,Type=Integer,Description="Total Depth",IDX=2>)"},
+            {R"(INFO)", R"(<ID=AF,Number=A,Type=Float,Description="Allele Frequency",IDX=3>)"},
+            {R"(INFO)", R"(<ID=AA,Number=1,Type=String,Description="Ancestral Allele",IDX=4>)"},
+            {R"(INFO)", R"(<ID=DB,Number=0,Type=Flag,Description="dbSNP membership, build 129",IDX=5>)"},
+            {R"(INFO)", R"(<ID=H2,Number=0,Type=Flag,Description="HapMap2 membership",IDX=6>)"},
+            {R"(FILTER)", R"(<ID=q10,Description="Quality below 10",IDX=7>)"},
+            {R"(FILTER)", R"(<ID=s50,Description="Less than 50% of samples have data",IDX=8>)"},
+            {R"(FORMAT)", R"(<ID=GT,Number=1,Type=String,Description="Genotype",IDX=9>)"},
+            {R"(FORMAT)", R"(<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality",IDX=10>)"},
+            {R"(FORMAT)", R"(<ID=DP,Number=1,Type=Integer,Description="Read Depth",IDX=2>)"},
+            {R"(FORMAT)", R"(<ID=HQ,Number=2,Type=Integer,Description="Haplotype Quality",IDX=11>)"},
+            {R"(bcftools_viewVersion)", R"(1.18+htslib-1.18)"},
+            {R"(bcftools_viewCommand)", R"(view -o vcf.bcf vcf.vcf; Date=Sun Nov 12 18:58:41 2023)"},
+        },
+        .genotypes = {"NA00001", "NA00002", "NA00003"},
+    };
+
+    auto test_data = []() -> std::string {
```

Comment:
Just as a note:

It's very helpful to write down how you got the binary output (especially with BAM).
Otherwise, one has to reinvent the wheel when the tests change.

We had this problem for BAM tests in seqan3.

After rediscovering how to generate the test binary input, we wrote it down:
https://github.com/seqan/seqan3/blob/ffabe95327e99f954db0cf116ce189b36284c49d/test/unit/io/sam_file/format_bam_test.cpp#L17-L30

Reply:
Yes, very good idea!